### ISSUE TITLE
[Doc]: Rewrite of image tutorial

### ISSUE THREAD

**Issue by philipmnelson** [matplotlib/matplotlib#22790]
### Documentation Link

https://matplotlib.org/devdocs/tutorials/introductory/images.html

### Problem

I found several issues within this introductory tutorial and I believe those issues can be solved by rewriting the tutorial.

**Issues:**
* First, this tutorial was written prior to the depreciation of matplotlib.image.imread(), the current preferred method of opening images is using [Pillow's Image.open()](https://matplotlib.org/devdocs/api/image_api.html#matplotlib.image.imread). The tutorial doesn't indicate this and reader's of the tutorial will see depreciation warnings when trying to go through the tutorial.
* Second, I believe this tutorial is quite lengthy for an intro on how to use matplotlib.pyplot.imshow(). I think it would be better not to overwhelm new users with paragraphs of information on image analysis techniques and theory. Linking to pages about these topics might save users time.

I am considering rewriting this tutorial but first I want to get an idea of how to rewrite it in a way that avoids adding too many details but also gives users a good introduction to imshow(). As this is the first time I've contributed here, I wanted to get input from others before I started this process.

**Comment by jklymak** [comment]
@philipmnelson I think this tutorial could well benefit from a fresh set of eyes and some de-crufting.  The reason our docs accrete extraneous info is usually that a user trips across some rough edge, we explain it to them, they ask where they could have figured this out themselves, and then someone updates the most relevant tutorial.  Over time, you paste enough of these asides on the tutorial and its hard to follow what is going on.  

Put the simple stuff first.  All the detail isn't bad, but as you say upfront can be very confusing.   Add many links - if there is a digression at the bottom that is important at the top, link to it at the top, but don't digress too much; if there is another place something explained link to it; if there are other tutorials or examples, link to them.  

You mention that we use deprecated imread, but Pillow doesn't normalize from 256 to 1;  maybe worth opening another issue about that.  I'm not sure if that was discussed when we decided to deprecate.  

Thanks for being willing to work on this.

**Comment by philipmnelson** [comment]
Okay, I just wanted to clarify for anybody who sees this issue that imread() is *not* deprecated, but the docs say that it exists only for historical reasons. Reading a URL with imread *is* deprecated and the alternative method provided in the reference docs and depreciation is not a complete replacement for imread's URL reading functionality.

**Comment by Bustead** [comment]
@blu3r4d0n if you want a hand let me know! I'm just starting out too.

**Comment by timhoffm** [comment]
@Bustead since this is inactive for over a year, feel free to move forward on your own if there isn't response within a few days.
